src/main/java/org/broadinstitute/consent/http/db/DataSetDAO.java

```diff
@@ -99,6 +99,17 @@
             "where d.dataSetId = :dataSetId order by d.dataSetId, k.displayOrder")
     Set<DataSetDTO> findDataSetWithPropertiesByDataSetId(@Bind("dataSetId") Integer dataSetId);
 
+  @UseRowMapper(DataSetPropertiesMapper.class)
+  @SqlQuery("SELECT d.*, k.key, dp.propertyvalue, ca.consentid, c.dac_id, c.translateduserestriction " +
+      "FROM dataset d " +
+      "LEFT OUTER JOIN datasetproperty dp on dp.datasetid = d.datasetid " +
+      "LEFT OUTER JOIN dictionary k on k.keyid = dp.propertykey " +
+      "LEFT OUTER JOIN consentassociations ca on ca.datasetid = d.datasetid " +
+      "LEFT OUTER JOIN consents c on c.consentid = ca.consentid " +
+      "WHERE d.datasetid = :datasetId " +
+      "ORDER BY d.datasetid, k.displayorder")
+  Set<DataSetDTO> findDatasetDTOWithPropsByDatasetId(@Bind("datasetId") Integer datasetId);
```

Comment:
I'm confused about why we're returning a set here. We know there's only one, why not just this?
```suggestion
  DataSetDTO findDatasetDTOWithPropsByDatasetId(@Bind("datasetId") Integer datasetId);
```

Reply:
So from what I can tell, the DataSetDTO is somewhat misleading of a name here. What is being returned by the DAO initially is a list of properties that DataSetPropertiesMapper then reformats into what we know as the DataSetDTO. This has to be a set so that we can grab all of the properties for a given dataset, rather than only the first property. It's pretty confusing but I'm not sure how to make it more obvious without separating the fetch of properties and dataset and tying them back together in the service instead? Would that be a better approach to using the mapper?

Reply:
If you look at `DataSetPropertiesMapper`, it creates a map of dataset id to DataSetDTO. It will only return a single DataSetDTO for a dataset id. If you queried on multiple different ids, then you'd get more than one back. The mapper will take all of the properties with the same dataset id, any number of them, and turn them into single DataSetDTO. Give my suggestion a shot and you should see all the resulting rows come back as a single object.